### PR TITLE
[FB-851] Update Redis Version

### PR DESCRIPTION
**Pull request by caesarprime**
Description of your patch
-------------
The patch updates the `redis` version of a redis instance included environment to the latest available version.

Recommended Release Notes
-------------
Applies `Security Updates` to Redis. Fixes the vulnerability `CVE-2019-10192`

Estimated risk
-------------
Medium - This involved a heap overflow vulnerability

Components involved
-------------
`redis` cookbook, `custom-redis` cookbook and `ey-custom` cookbook

Description of testing done
-------------

**Redis Cookbook

* Booted an environment with a Ruby app , Database and a redis util instance and a memcached instance
* checked whether redis has got any new updates by connecting to the redis instance (`emerge -upDN1 world` or `eix dev-db/redis`)
* Imported `redis` via `depends 'redis'` on `ey-custom/metadata.rb`, and added `import_recipe 'redis::default'` to `ey-custom/recipes/after_main.rb`
* Executed `/home/ey/bin/chef-solo -j /etc/chef/dna.json -c /etc/chef/solo.rb`
* Checked `redis` for the latest available version - `dev-db/redis` and for 'monit summary` to check that monit is running for `redis`

**Redis Custom Cookbook

* Booted an environment with a Ruby app , Database and a redis util instance and a memcached instance
* checked whether redis has got any new updates by connecting to the redis instance (`emerge -upDN1 world` or `eix dev-db/redis`)
* Copied the `redis-custom` cookbook to `cookbooks` directory
* Imported `redis-custom` via `depends 'redis-custom'` on `ey-custom/metadata.rb`, and added `import_recipe 'custom-redis::default'` to `ey-custom/recipes/after_main.rb`
* Ran `ey-core recipes upload --environment <environment-name>`
* Run apply on environment dashboard
* Checked `redis` for the latest available version - `dev-db/redis`

QA Instructions
-------------
* Test a PHP Application
* Test with a single instance env
* Test with a multiple instance env without redis instance
* Test with different config envs with redis instance

### CODE REVIEW

File: cookbooks/redis/attributes/overrides.rb [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Overrides default Redis version with the latest available
```
Comment (jfuechsl) [inline cookbooks/redis/attributes/overrides.rb]: Why are you creating this file? It's sufficient to set the version in `cookbooks/redis/attributes/default.rb`

File: custom-cookbooks/redis/cookbooks/custom-redis/attributes/overrides.rb [hunk 2]
```diff
@@ -0,0 +1,2 @@
+# FB-851
```
Comment (jfuechsl) [inline custom-cookbooks/redis/cookbooks/custom-redis/attributes/overrides.rb]: Ditto here. It's already updated in `custom-cookbooks/redis/cookbooks/custom-redis/attributes/default.rb`

File: cookbooks/redis/attributes/overrides.rb (hunk 1 above)
Comment (caesarprime) [reply cookbooks/redis/attributes/overrides.rb]: Just thought to arrange them up with abstracting some most changeable features. But let me remove that file and move with only the `default.rb`

File: custom-cookbooks/redis/cookbooks/custom-redis/attributes/overrides.rb (hunk 2 above)
Comment (caesarprime) [reply custom-cookbooks/redis/cookbooks/custom-redis/attributes/overrides.rb]: Yep true, I'll be removing the `overrides.rb` via the next release.

**Pull request merged**